src/main/webapp/js/user.js

```diff
@@ -0,0 +1,91 @@
+function autocomplete(input, languages) {
+  let currentFocus;
+  input.addEventListener('input', function(e) {
+    let autocomplete = this.value;
+    let language = this.value;
+    const val = this.value;
+    closeAllLists();
+    if (!val) {
+      return false;
+    }
+    currentFocus = -1;
+    autocomplete = document.createElement('div');
+    autocomplete.setAttribute('id', this.id + 'autocomplete-list');
+    autocomplete.setAttribute('class', 'autocomplete-items');
+    this.parentNode.appendChild(autocomplete);
+    for (let i = 0; i < languages.length; i++) {
+      if (languages[i].substr(0,
+          val.length).toUpperCase() == val.toUpperCase()) {
+        language = document.createElement('div');
+        // Add each element with bolded part for input characters.
+        language.innerHTML = '<strong>' + languages[i].substr(0,
+            val.length) + '</strong>' + languages[i].substr(val.length);
+        language.innerHTML += '<input type="hidden" value="'+
+            languages[i] + '">';
+        language.addEventListener('click', function(e) {
+          input.value = this.getElementsByTagName('input')[0].value;
+          closeAllLists();
+        });
+        autocomplete.appendChild(language);
+      }
+    }
+  });
+  /* Execute a function presses a key on the keyboard */
+  input.addEventListener('keydown', function(e) {
```

Comment:
Implementing keyboard accessibility is a nice touch, but it's hard to do. There's a bunch of bugs I see here. For one, you can press the down arrow past the visible options.

Also note that for actual accessibility, you should be changing what element has focus as you press keys. Then you don't need to handle Enter (and Space, which was omitted, will work just like that). This also makes is so that the event `e` tells you what element was previously focused, so you don't have to deal with all this autocomplete-active maintenance.